features/step_definitions/upgrade/etcd.rb

```diff
@@ -75,4 +75,78 @@
   logger.info("### etcdCluster: #{name} is removed successfully from #{proj_name} namespace")
 end
 
+Then /^(the|all)? etcd pods outputs?( by order)? should( not)? (contain|match)? #{NUMBER} times:$/ do |all, in_order, negative, match_type, num, table|
```

Comment:
@geliu2016 @sandeepknd this step is unclear and perhaps over complicated.   For example what does the following mean?  ```And the etcd pods output should contain <%= cb.masters.length %> times```  What is the pattern it's watching over?

Reply:
@pruan-rht this step will check etcd member/guard pod, which in "Running" status, if there are 3 member pods, so I want to get 3 Running, if there is 1 member pods in SNO, then I want to get 1 Running

Reply:
@geliu2016 ok...I would rephrase the step to make the purpose more clear.  Also, I would simplify the things to check to lessen the logic.   For example, the  (by order, and contain|match) added more logic to the step that don't really bring much more values.

Reply:
@pruan-rht Great, thanks in advance

Reply:
@geliu2016 @sandeepknd after reviewing your testcase, you can simplifies by a lot by using existing methods.  I'll comment more inline... This new step is not needed.